pkg/controller/jaeger/jaeger_controller.go

```diff
@@ -222,3 +232,53 @@ func (r *ReconcileJaeger) apply(jaeger v1.Jaeger, str strategy.S) error {
 
 	return nil
 }
+
+//addFinalizer will add this attribute to the Jaeger CR
+func (r *ReconcileJaeger) addFinalizer(jaeger *v1.Jaeger) error {
+	if len(jaeger.GetFinalizers()) < 1 && jaeger.GetDeletionTimestamp() == nil {
+		jaeger.Logger().Info("Adding Finalizer")
+		jaeger.SetFinalizers([]string{finalizer})
+		// Update CR
+		if err := r.client.Update(context.Background(), jaeger); err != nil {
+			jaeger.Logger().Error(err, "Failed to update Jaeger with finalizer")
+			return err
+		}
+	}
+	return nil
+}
+
+func (r *ReconcileJaeger) clean(jaeger *v1.Jaeger) error {
+	opts := client.MatchingLabels(map[string]string{
+		inject.Label: jaeger.Name,
+	})
+	depList := &appsv1.DeploymentList{}
+	if err := r.client.List(context.Background(), opts, depList); err != nil {
+		return err
+	}
+	inject.CleanSidecars(depList.Items)
+	for _, d := range depList.Items {
+		if err := r.client.Update(context.Background(), &d); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (r *ReconcileJaeger) handleFinalizer(instance *v1.Jaeger) (bool, error) {
```

Comment:
Do you need to return bool from this method? It does not seem to be used in the caller.

Reply:
mmh yeah not necessary any more, I think I used for something but not necessary now I'll remove it.